modules/integration/tests-integration/tests-mqtt/src/test/java/org/wso2/mb/integration/tests/mqtt/functional/BasicSendCompressReceiveTestCase.java

```diff
@@ -0,0 +1,139 @@
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class BasicSendCompressReceiveTestCase extends MBIntegrationBaseTest {
+
+    /**
+     * Initialize super class.
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    public void prepare() throws Exception {
+        super.init(TestUserMode.SUPER_TENANT_USER);
+    }
+
+    /**
+     * Set allowCompression to true so that broker will compress messages before storing into the database, and
+     * reduce maximum content chunk size to 100 from default value (65500); to create more content chunks
+     * from compressed content, to check chunk data retrieval.
+     *
+     * @throws XPathExpressionException
+     * @throws java.io.IOException
+     * @throws org.apache.commons.configuration.ConfigurationException
+     * @throws org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException
+     */
+    @BeforeClass
+    public void setupConfiguration() throws XPathExpressionException, IOException, ConfigurationException,
+            AutomationUtilException {
+
+        super.serverManager = new ServerConfigurationManager(automationContext);
+        String defaultMBConfigurationPath = ServerConfigurationManager.getCarbonHome() +
+                File.separator + "repository" + File.separator + "conf" + File.separator + "broker.xml";
+
+        ConfigurationEditor configurationEditor = new ConfigurationEditor(defaultMBConfigurationPath);
+
+        configurationEditor.updateProperty(AndesConfiguration.PERFORMANCE_TUNING_ALLOW_COMPRESSION, "true");
+        configurationEditor.updateProperty(AndesConfiguration.PERFORMANCE_TUNING_MAX_CONTENT_CHUNK_SIZE, "100");
+
+        configurationEditor.applyUpdatedConfigurationAndRestartServer(serverManager);
+    }
+
+    /**
+     * Send a single mqtt message on qos {@link QualityOfService#LEAST_ONCE} and receive.
+     *
+     * @throws MqttException
+     */
+    @Test(groups = {"wso2.mb", "mqtt"}, description = "Single mqtt message send compress receive test case")
+    public void performBasicSendCompressReceiveTestCase() throws MqttException, XPathExpressionException,
+            IOException {
+
+        String topic = "BasicSendCompressReceiveTestCase";
+        int noOfSubscribers = 1;
+        int noOfPublishers = 1;
+        int noOfMessages = 1;
+        boolean saveMessages = true;
+        //Input file size is 256KB
+        int messageSize = 250 * 1024;
+        String filePath = MQTTConstants.MESSAGE_CONTENT_RANDOM_FILE_PATH;
+
+        MQTTClientEngine mqttClientEngine = new MQTTClientEngine();
+        //create the subscribers
+        mqttClientEngine.createSubscriberConnection(topic, QualityOfService.LEAST_ONCE, noOfSubscribers, saveMessages,
+                ClientMode.BLOCKING, automationContext);
+
+        AndesClientUtils.createRandomFile(filePath, messageSize);
```

Comment:
Instead of creating a file,re-reading it and getting byte[] can't we get the byte [] directly from the function?